js/services.ts

```diff
@@ -135,11 +139,130 @@ module jsonforms.services {
 
     }
 
+    export class UISchemaGenerator{
+        generateDefaultUISchema = (jsonSchema:any):any =>{
+            var uiSchemaElements = [];
+            this.generateUISchema(jsonSchema, uiSchemaElements, "#", "");
+
+            console.log("generated schema: " + JSON.stringify(uiSchemaElements[0]))
+
+            return uiSchemaElements[0];
+        };
+
+        private generateUISchema = (jsonSchema:any, schemaElements:IUISchemaElement[], currentRef:string, schemaName:string):any =>{
+            if(!jsonSchema.type){
```

Comment:
As a side note: JSON schema does not necessarily require the type keyword to be defined on a schema, for instance, the presence of the properties keyword suffices in order to derive the object type.